source/kraken/apply_disruption.cpp

```diff
@@ -407,7 +412,12 @@ struct add_impacts_visitor : public apply_impacts_visitor {
 
                 // for protection, use the datasets[0]
                 // TODO : Create default data set
-                vj->dataset = pt_data.datasets[0];
+                if (!pt_data.datasets.empty()) {
+                    if (pt_data.datasets[0]) {
```

Comment:
Looks like you're checking almost twice the same thing, but if you want to keep double check it's probably more expressive like:
```suggestion
                if (!pt_data.datasets.empty() && pt_data.datasets[0] != nullptr) {
```